docroot/modules/custom/mass_utility/src/Commands/MassUtilityCommands.php

```diff
@@ -398,4 +399,95 @@ public function queueRevisionsCleanup(array $options = [
 
   }
 
+  /**
+   * Copy D2D nodes into redirect entities.
+   *
+   * @command copyd2d
+   * @option limit
+   * @option min-hits
+   */
+  public function copyD2DRedirects($options = ['limit' => self::REQ, 'min-hits' => 50]) {
+    $query = $this->database->select('node__field_legacy_redirects_legacyurl', 'source')
+      ->fields('source', ['entity_id', 'delta']);
+    $query->addField('source', 'field_legacy_redirects_legacyurl_value', 'source');
+    $query->addField('target', 'field_legacy_redirects_ref_conte_target_id', 'target');
+    if ($options['limit']) {
+      $query->range(0, $options['limit']);
+    }
+    $query->innerJoin('node__field_legacy_redirects_ref_conte', 'target', 'source.entity_id = target.entity_id AND source.delta = target.delta');
```

Comment:
Hmm, I think the delta condition here is unwanted.  My understanding is that Legacy redirects can have many sources, but only one target per node.  Meaning that every source redirects to target delta = 0.

Reply:
I think that this query wants to load of pairs of source and target from the D2D content type. So we do want to make sure we match deltas. If you aren't convinced, lets discuss. This area is a bit hazy for me too.

Reply:
I don't think that's right - here's a screenshot of a crazy example of the source field:
![Screen Shot 2020-03-03 at 5 27 06 PM](https://user-images.githubusercontent.com/654407/75825968-72b0cb80-5d74-11ea-8154-7643d2468318.png)
Meanwhile the destination field is a single value:
![Screen Shot 2020-03-03 at 5 29 04 PM](https://user-images.githubusercontent.com/654407/75826023-88be8c00-5d74-11ea-99cc-907e526fcba3.png)